aspnetcore/fundamentals/minimal-apis/samples/WebMinAPIs/Program.cs

```diff
@@ -813,4 +813,20 @@ class Product
 app.Run();
 #endregion
 
+#elif FILEUPLOAD
+#region snippet_fileupload
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapPost("/uploadfile", async (IConfiguration config, HttpRequest request) =>
+{
+    var filePath = Path.Combine(config["StoredFilesPath"], Path.GetRandomFileName());
+
+    await using var writeStream = File.Create(filePath);
+    await request.Body.CopyToAsync(writeStream);
```

Comment:
What do we think about using the `BodyReader` to avoid a copy?

```suggestion
    await request.BodyReader.CopyToAsync(writeStream);
```